src/libraries/StackUint256.sol

```diff
@@ -0,0 +1,56 @@
+pragma solidity ^0.4.13;
+
+import "ROOT/libraries/DelegationTarget.sol";
+import "ROOT/legacy_reputation/Ownable.sol";
+import "ROOT/libraries/Delegator.sol";
+import "ROOT/Controller.sol";
+
+
+contract StackUint256 is DelegationTarget, Ownable {
+    
+    uint256[] private collection;
+    uint256 private head;
```

Comment:
We no longer need `head` in Solidity.  We can use `collection.length` in its place.  For pop, you will need to do something like:
```
function pop() {
	require(collection.length > 0);
	uint256 poppedItem = collection[collection.length - 1];
	delete collection[collection.length - 1];
	collection.length -= 1;
	return poppedItem;
```

For `push` we can just remove the `head += 1` line.

For `peek` we would just use `return collection[collection.length - 1];`

Reply:
The same change can likely be applied to many of the other collections.  Anywhere where we were tracking the "most recently inserted item".  In some cases (like linked lists) though, `head` actually means head of a linked list and is different from the collection length.

Reply:
Actually, I just glanced through and it doesn't look like any of the other collections have a counter.  So just this one should change.